systemtest: ensure geoIP database is available before running tests

The system tests intermittently fail, due to the geoIP database being unavailable. e.g. https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/PR-7257/2/testReport/

```
=== RUN   TestErrorIngest
    errors_test.go:52:   map[string]interface{}{
          	"events": []interface{}{
          		map[string]interface{}{
          			... // 16 identical entries
          			"processor": map[string]interface{}{"event": string("error"), "name": string("error")},
          			"service":   map[string]interface{}{"environment": string("staging"), "framework": map[string]interface{}{"name": string("Express"), "version": string("1.2.3")}, "language": map[string]interface{}{"name": string("ecmascript"), "version": string("8")}, "name": string("1234_service-12a3"), ...},
        + 			"tags":      []interface{}{string("_geoip_database_unavailable_GeoLite2-City.mmdb")},
          			"timestamp": map[string]interface{}{"us": float64(1.533827045999e+15)},
          			"trace":     map[string]interface{}{"id": string("0123456789abcdeffedcba0123456789")},
          			... // 2 identical entries
          		},
          	},
          }
        
        Run `make update check-approvals` to verify the diff
        
    server.go:191: log file: /var/lib/jenkins/workspace/pm-server_apm-server-mbp_PR-7257/src/github.com/elastic/apm-server/systemtest/logs/TestErrorIngest/apm-server
--- FAIL: TestErrorIngest (1.39s)
```

We should ensure the geoIP database is available before running the tests. I'm not sure how yet. Maybe we can use the [simulate pipeline API](https://www.elastic.co/guide/en/elasticsearch/reference/current/simulate-pipeline-api.html) to force a download, and keep trying until it succeeds?

--- a/systemtest/containers.go
+++ b/systemtest/containers.go
@@ -50,10 +50,12 @@ import (
 	"github.com/elastic/apm-server/systemtest/apmservertest"
 	"github.com/elastic/apm-server/systemtest/estest"
 	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
 )
 
 const (
 	startContainersTimeout = 5 * time.Minute
+	waitHealthyInterval    = 5 * time.Second
 )
 
 var (
@@ -110,6 +112,7 @@ func StartStackContainers() error {
 	g, ctx := errgroup.WithContext(ctx)
 	g.Go(func() error { return waitContainerHealthy(ctx, "kibana") })
 	g.Go(func() error { return waitContainerHealthy(ctx, "fleet-server") })
+	g.Go(func() error { return waitGeoIPDatabase(ctx) })
 	return g.Wait()
 }
 
@@ -191,9 +194,49 @@ func waitContainerHealthy(ctx context.Context, serviceName string) error {
 			log.Printf("Waiting for %s container (%s) to become healthy", serviceName, container.ID)
 			first = false
 		}
-		time.Sleep(5 * time.Second)
+		time.Sleep(waitHealthyInterval)
 	}
 	return nil
+}
+
+// Elasticsearch downloads the geoIP database after starting up, and then refreshes it
+// periodically. In CI the cluster will be fresh, so we wait for the database to be
+// downloaded before running any tests which may depend on it being available.
+func waitGeoIPDatabase(ctx context.Context) error {
+	type geoIPDatabase struct {
+		Name string
+	}
+	type nodeGeoIPStats struct {
+		Databases []geoIPDatabase
+	}
+
+	first := true
+	for {
+		var out struct {
+			Nodes map[string]nodeGeoIPStats
+		}
+		_, err := Elasticsearch.Do(ctx, esapi.IngestGeoIPStatsRequest{}, &out)
+		if err != nil {
+			return err
+		}
+
+		if n := len(out.Nodes); n != 1 {
+			return fmt.Errorf("expected 1 node, got %d", n)
+		}
+		for _, nodeStats := range out.Nodes {
+			for _, db := range nodeStats.Databases {
+				if db.Name == "GeoLite2-City.mmdb" {
+					return nil
+				}
+			}
+		}
+
+		if first {
+			log.Printf("Waiting for geoIP database to be downloaded")
+			first = false
+		}
+		time.Sleep(waitHealthyInterval)
+	}
 }
 
 func stackContainerInfo(ctx context.Context, docker *client.Client, name string) (*types.Container, error) {

